src/common-pages/Subheader.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { AppBar, Toolbar, IconButton, Typography } from '@mui/material';
+import PersonIcon from '@mui/icons-material/Person';
+
+const Subheader: React.FC = () => {
+    const [activeItem, setActiveItem] = useState<string | null>(null);
+    const [isMobile, setIsMobile] = useState<boolean>(window.innerWidth <= 768);
+    const [showUserName, setShowUserName] = useState<boolean>(false);
+    const [appBarHeight] = useState<number>(80);
+    const mobileAppBarHeight = 100;
+
+    const handleResize = useCallback(() => {
+        setIsMobile(window.innerWidth <= 768);
+    }, []);
+
+    useEffect(() => {
+        window.addEventListener('resize', handleResize);
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
+    }, [handleResize]);
+
+    const handleMenuClick = (item: string) => {
+        setActiveItem(item);
+        window.history.pushState(null, '', `/${item}`);
```

Comment:
I belive this strategy does not work, take a look of https://mui.com/material-ui/react-link/